### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by supriyopaul**

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2018 supriyopaul
```
Comment (jaswanth098) [inline LICENSE]: Same mistake again @supriyopaul

File: nsq_api/service.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+            topic, channel))
+
+    @gen.coroutine
+    def get(self):
+        '''
+        Sample uri: /tail?nsqd_tcp_address=localhost:4150&topic=Heartbeat&empty_lines='yes/no'
+        '''
+        loop = tornado.ioloop.IOLoop.current()
+
+        nsqd_tcp_addresses, topic, empty_lines = self._parse_query(self.request.query_arguments)
```
Comment (jaswanth098) [inline nsq_api/service.py]: @supriyopaul This one will throw an error if someone making a request with less than three parameters please ensure that we have all the parameters that are required before doing something like this.

File: nsq_api/service.py [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+    def initialize(self, log):
+
+        self.log = log
+
+    def _parse_query(self, query_arguments):
+        '''
+        query_arguments: {'topic': ['Heartbeat'], 'nsqd_tcp_address': ['195.201.98.142:4150']}
+        '''
+
+        nsqd_tcp_addresses=query_arguments['nsqd_tcp_address']
```
Comment (jaswanth098) [inline nsq_api/service.py]: what if there is no nsqd_tcp_address in query_arguents?

File: nsq_api/service.py [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+    DELETE_CHANNEL_URL = 'http://{}/channel/delete?topic={}&channel={}'
+    executor = concurrent.futures.ThreadPoolExecutor(5)
+
+    def initialize(self, log):
+
+        self.log = log
+
+    def _parse_query(self, query_arguments):
+        '''
+        query_arguments: {'topic': ['Heartbeat'], 'nsqd_tcp_address': ['195.201.98.142:4150']}
```
Comment (jaswanth098) [inline nsq_api/service.py]: I don't think we should hardcode the IP address here.

File: nsq_api/service.py [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+
+        return nsqd_tcp_addresses, topic, empty_lines
+
+    def _remove_channel(self, nsqd_host,
+            topic, channel):
+        '''
+        Delete a Nsq channel
+        '''
+
+        nsqd_http_port = '4151'
```
Comment (jaswanth098) [inline nsq_api/service.py]: Shouldn't the port be configurable?

File: nsq_api/service.py (hunk 4 above)
Comment (supriyopaul) [reply nsq_api/service.py]: It's just a comment with arbitrary IP address.

File: nsq_api/service.py (hunk 2 above)
Comment (supriyopaul) [reply nsq_api/service.py]: And if all the query parameters are not there, what should I do?

Comment (jaswanth098) [reply nsq_api/service.py]: return back saying required parameters are missing

Comment (supriyopaul) [reply nsq_api/service.py]: But the user is expecting a straming response, It will cause errors at client side then.

Comment (jaswanth098) [reply nsq_api/service.py]: If user doesn't pass the parameters needed how will you stream the data

Comment (supriyopaul) [reply nsq_api/service.py]: Streaming does not allow to put returns in the function:
    `SyntaxError: 'return' with argument inside generator`
Plus client gets an internal server error, so it doesn't fail silently.

Comment (supriyopaul) [reply nsq_api/service.py]: And if someone uses the url wrong why is it the API's responsibility to correct it? I agree a proper error message but it'l be too much of a complexity for such a small problem.

File: nsq_api/service.py [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+
+    def initialize(self, log):
+
+        self.log = log
+
+    def _parse_query(self, query_arguments):
+        '''
+        query_arguments: {'topic': ['Heartbeat'], 'nsqd_tcp_address': ['195.201.98.142:4150']}
+        '''
+        nsqd_tcp_addresses = query_arguments.get('nsqd_tcp_address', [''])[0]
```
Comment (jaswanth098) [inline nsq_api/service.py]: the default cannot be used in if condition 
```python
>>> if [""]:
...   print "hello"
... 
hello
```

Comment (jaswanth098) [inline nsq_api/service.py]: We can directly empty string here instead of taking `0` element from list

Comment (supriyopaul) [reply nsq_api/service.py]: But query args give a list of strings.
`nsqd_tcp_addresses  = nsqd_tcp_addresses [0]` will be an additional step after parsing.

**Pull request merged**